### ISSUE TITLE
🐛 Bug Report: Unauthorized from Bitbucket Cloud

### ISSUE THREAD

**Issue by knewman23** [backstage/backstage#22846]
### 📜 Description

I have Bitbucket Cloud configured as a provider and integration. I'm able to open the search url in the browser and see a json response, but when I run from backstage it throws a 401

```
integrations:
  bitbucketCloud:
    - username: <redacted>
      appPassword: <redacted>
```

```
catalog:
  providers:
    bitbucketCloud:
      brivoinc: # identifies your ingested dataset
        catalogPath: /catalog-info.yaml # default value
        schedule: # same options as in TaskScheduleDefinition
          frequency: { minutes: 2 }
          timeout: { minutes: 3 }
        workspace: brivoinc
```

### 👍 Expected behavior

Authorize and pull repositories

### 👎 Actual Behavior with Screenshots

Errors out in the console

```
[1] 2024-02-09T06:51:06.197Z catalog error bitbucketCloud-provider:brivoinc refresh failed, Error: Unexpected response for GET https://api.bitbucket.org/2.0/workspaces/brivoinc/search/code?page=1&pagelen=100&fields=-values.content_matches%2C%2Bvalues.file.commit.repository.mainbranch.name%2C%2Bvalues.file.commit.repository.project.key%2C%2Bvalues.file.commit.repository.slug%2C-values.*.links%2C-values.*.*.links%2C-values.*.*.*.links%2C%2Bvalues.file.commit.repository.links.html.href&search_query=%22catalog-info.yaml%22+path%3A%2Fcatalog-info.yaml. Expected 200 but got 401 - Unauthorized Unexpected response for GET https://api.bitbucket.org/2.0/workspaces/brivoinc/search/code?page=1&pagelen=100&fields=-values.content_matches%2C%2Bvalues.file.commit.repository.mainbranch.name%2C%2Bvalues.file.commit.repository.project.key%2C%2Bvalues.file.commit.repository.slug%2C-values.*.links%2C-values.*.*.links%2C-values.*.*.*.links%2C%2Bvalues.file.commit.repository.links.html.href&search_query=%22catalog-info.yaml%22+path%3A%2Fcatalog-info.yaml. Expected 200 but got 401 - Unauthorized type=plugin target=bitbucketCloud-provider:brivoinc class=BitbucketCloudEntityProvider taskId=bitbucketCloud-provider:brivoinc:refresh taskInstanceId=3697d37f-5161-48de-bd77-8f076d891169 stack=Error: Unexpected response for GET https://api.bitbucket.org/2.0/workspaces/brivoinc/search/code?page=1&pagelen=100&fields=-values.content_matches%2C%2Bvalues.file.commit.repository.mainbranch.name%2C%2Bvalues.file.commit.repository.project.key%2C%2Bvalues.file.commit.repository.slug%2C-values.*.links%2C-values.*.*.links%2C-values.*.*.*.links%2C%2Bvalues.file.commit.repository.links.html.href&search_query=%22catalog-info.yaml%22+path%3A%2Fcatalog-info.yaml. Expected 200 but got 401 - Unauthorized
[1]     at <anonymous> (/Users/krysnewman/Projects/brivo/node_modules/@backstage/plugin-bitbucket-cloud-common/src/BitbucketCloudClient.ts:97:17)
[1]     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
[1]     at WithPagination.iterateResults (/Users/krysnewman/Projects/brivo/node_modules/@backstage/plugin-bitbucket-cloud-common/src/pagination.ts:59:13)
[1]     at BitbucketCloudEntityProvider.findCatalogFiles (/Users/krysnewman/Projects/brivo/node_modules/@backstage/plugin-catalog-backend-module-bitbucket-cloud/src/providers/BitbucketCloudEntityProvider.ts:361:22)
[1]     at BitbucketCloudEntityProvider.refresh (/Users/krysnewman/Projects/brivo/node_modules/@backstage/plugin-catalog-backend-module-bitbucket-cloud/src/providers/BitbucketCloudEntityProvider.ts:188:21)
[1]     at fn (/Users/krysnewman/Projects/brivo/node_modules/@backstage/plugin-catalog-backend-module-bitbucket-cloud/src/providers/BitbucketCloudEntityProvider.ts:153:19)
[1]     at TaskWorker.fn (/Users/krysnewman/Projects/brivo/node_modules/@backstage/backend-tasks/src/tasks/PluginTaskSchedulerImpl.ts:134:9)
[1]     at TaskWorker.runOnce (/Users/krysnewman/Projects/brivo/node_modules/@backstage/backend-tasks/src/tasks/TaskWorker.ts:155:13)
[1]     at <anonymous> (/Users/krysnewman/Projects/brivo/node_modules/@backstage/backend-tasks/src/tasks/TaskWorker.ts:75:31)
```

### 👟 Reproduction steps

```
integrations:
  bitbucketCloud:
    - username: <redacted>
      appPassword: <redacted>
```

```
catalog:
  providers:
    bitbucketCloud:
      brivoinc: # identifies your ingested dataset
        catalogPath: /catalog-info.yaml # default value
        schedule: # same options as in TaskScheduleDefinition
          frequency: { minutes: 2 }
          timeout: { minutes: 3 }
        workspace: brivoinc
```

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

**Comment by benjdlambert** [comment]
@pjungermann does anything stand out here, maybe you might have some knowledge about the `username`/ `appPassword` combo?

**Comment by pjungermann** [comment]
@knewman23 Which permissions did you give to the credentials?

**Comment by knewman23** [comment]
@pjungermann I gave full permissions to the app token. the initial startup says `bitbucketCloud{host=bitbucket.org,authed=true}` but our bitbucket instance is tied to okta sso for login, would it be required to set up an okta login integration for this case?

**Comment by pjungermann** [comment]
Our setup looks similar:

```yaml
integrations:
  bitbucketCloud:
    - appPassword: ${BITBUCKET_PASSWORD}
      username: ${BITBUCKET_USERNAME}
```

```yaml
catalog:
  providers:
    bitbucketCloud:
      some-name:
        catalogPath: catalog-info.yaml # in any dir
        filters:
          repoSlug: '^some-regex$'
        schedule:
          frequency: { cron: '0 10 * * SUN' }
          timeout: { minutes: 3 }
        workspace: workspace-slug
```

The permissions provided to the app password cannot exceed the ones from the related user, too.

Did you try using the same credentials with the APIs?

In particular the code search API `/workspaces/{workspace}/search/code` (see also `BitbucketCloudClient.searchCode` from `@backstage/plugin-bitbucket-cloud-common`).

Maybe, that way you can verify whether the credentials are to blame or not.

**Comment by MehulChamria** [comment]
I have the exact same issue and it start from 16th Feb just before midnight. No permissions were changed and it worked perfectly fine before that. Has anyone found a solution yet?

**Comment by pjungermann** [comment]
@knewman23 Did you have a chance to look further into your issue and the used secret and its permissions?

**Comment by benjdlambert** [comment]
@monkeyhandz I'm pretty confident that the issue you're seeing will be fixed in this `patch` release that we just shipped: https://github.com/backstage/backstage/pull/24496

Can you try update to `v1.26.4`? :pray:

**Comment by monkeyhandz** [comment]
> @monkeyhandz I'm pretty confident that the issue you're seeing will be fixed in this `patch` release that we just shipped: #24496
> 
> Can you try update to `v1.26.4`? 🙏
> 
> Your issue isn't really to do with BitBucket auth, rather it's Backstage's auth instead.

Thanks @benjdlambert upgrading works. I did expect my bitbucket repositories to be automatically imported, but I now understand I need to add a `catalog-info.yaml` to each repository.  Unless there is a way to automatically import them?

**Comment by knewman23** [comment]
Thank you for the update, I was able to get authenticated and pull repos!
